CIP-0118/README.md

```diff
@@ -0,0 +1,647 @@
+transaction constructed for them by the SP to do, e.g. pay `x` amount from my 
+wallet to some other wallet, or mint some NFT. 
+
+1. LC -> SP : a high-level specification `s` 
+2. SP : construct a `tx` that 
+- meets `s`, 
+- includes one input `i` from SP's wallet
+- includes a payment to SP for light-client services
+3. SP : replace all inputs except `i` in `tx` with corresponding `spendOuts` from the UTxO set to make `tx'`
+4. SP -> LC : `tx'`
+5. LC : checks `tx'` meets `s`, signs it
+6. LC -> SP : signed `tx'`
+7. SP : constructs top-level `tx''` that provides the `corInputs` for `tx'`
+8. SP : submits to Cardano the top-level `tx''` with sub-tx `tx'`
+
+Note that `tx'` may also be unbalanced, simultaneously allowing the LC 
+to make a payment in a non-Ada token for the service. This design ensures that 
+an LC cannot construct a fully valid transaction from `tx'`, since they do not have access to 
+the current UTxO (presumably, if they did, they would construct their own valid transaction).
+It also ensures that the SP is paid (in step 2) for their services. 
+
+## Rationale: how does this CIP achieve its goals?
+
+The primary purpose of this CIP is to enable Babel Fees. Our design allows users to perform arbitrary 
+swaps without two-way off-chain communication, and with counterparty irrelevance. This simulates
+fee payment in arbitrary currency whenever another use is willing 
+to accept the non-primary asset offer in exchange for paying the transaction fee. This is the core 
+requirement of Babel fees. In addition, the goal of this design is to lay the groundwork for 
+supporting more general intents and the possibility of extending to additional ones. 
+We have done so by adding the following set of features :
+
+1. transactions can be batched
+- batch contains one top-level transaction, with multiple possible sub-transactions ;
+
+2. the batch must be balanced, but individual transactions in it do not have to be, so 
+- transactions in a single batch may pay each others' fees and minUTxOValue ;
+- unbalanced transactions can be interpreted as swap offers, getting resolved within a complete batch ;
+
+3. script data is shared across all transactions in a single batch 
+- enables script deduplication ;
+
+4. batch observer scripts enable sub-transactions to specify properties required of batches in which they may be included 
+- design inspired by script observers ;
+- represents all possible intents that do not require ledger rule relaxation ;
+
+5. the top-level transaction must provide collateral for all scripts in its sub-transactions 
+- sub-transactions are not required to cover their own collateral ;
+
+6. a new intent we call "spend-by-output", wherein a sub-transaction may specify outputs it 
+- intends to spend, and the top-level transaction specifies the inputs that point to those outputs in the UTxO set ;
+- example of possible additional supported intents (see CPS-15)
```

Comment:
I am really sorry, but I have to push back on this feature, since it is fairly complex and is unclear to me. Which is a really good indicator that it needs its own CIP.
Nested transactions in itself is already a complex beast, we don't want to bloat it more than necessary. Since we'll have to implemented, I would for us to keep our sanity.

FYI, it would be nice to have a quote or title of the section from CPS-15 that talks about this use case.